plugins/modules/ipaautomember.py

```diff
@@ -0,0 +1,314 @@
+    remove_diff = set(current_conditions) - set(module_conditions)
+
+    for item in add_diff:
+        key, condition = item.split("=")
+        condition_args = gen_condition_args(grouping, key, condition, None)
+        commands.append([name, 'automember_add_condition', condition_args])
+
+    for item in remove_diff:
+        key, condition = item.split("=")
+        condition_args = gen_condition_args(grouping, key, condition, None)
+        commands.append([name, 'automember_remove_condition', condition_args])
+
+    return commands
+
+
+def main():
+    ansible_module = AnsibleModule(
+        argument_spec=dict(
+            # general
+            ipaadmin_principal=dict(type="str", default="admin"),
+            ipaadmin_password=dict(type="str", required=False, no_log=True),
+
+            inclusive=dict(type="list", aliases=[
+                           "automemberinclusiveregex"], default=None),
+            exclusive=dict(type="list", aliases=[
+                           "automemberexclusiveregex"], default=None),
+            name=dict(type="list", aliases=["cn"],
+                      default=None, required=True),
+            description=dict(type="str", default=None),
+            type=dict(type='str', required=True,
+                      choices=['group', 'hostgroup']),
+            state=dict(type="str", default="present",
+                       choices=["present", "absent"]),
+        ),
+        supports_check_mode=True,
+    )
+
+    ansible_module._ansible_debug = True
+
+    # Get parameters
+
+    # general
+    ipaadmin_principal = ansible_module.params.get("ipaadmin_principal")
+    ipaadmin_password = ansible_module.params.get("ipaadmin_password")
+    names = ansible_module.params.get("name")
+
+    # present
+    description = ansible_module.params.get("description")
+
+    # conditions
+    inclusive = ansible_module.params.get("inclusive")
+    exclusive = ansible_module.params.get("exclusive")
+
+    # state
+    state = ansible_module.params.get("state")
+
+    # grouping/type
+    grouping = ansible_module.params.get("type")
+
+    # Init
+    changed = False
+    exit_args = {}
+    ccache_dir = None
+    ccache_name = None
+    res_find = None
+
+    try:
+        if not valid_creds(ansible_module, ipaadmin_principal):
+            ccache_dir, ccache_name = temp_kinit(ipaadmin_principal,
+                                                 ipaadmin_password)
+        api_connect()
+
+        commands = []
+
+        for name in names:
+            # Make sure automember rule exists
+            res_find = find_automember(ansible_module, name, grouping)
+
+            # Create command
+            if state == 'present':
+                args = gen_args(description, grouping)
+
+                if res_find is not None:
+                    if not compare_args_ipa(ansible_module,
+                                            args, res_find, ['type']):
```

Comment:
Why is it needed to ignore `type`? What is it used for and what is the result of ignoring `type` also? In which case is this used?

Reply:
Morning @t-woerner thanks for your review.. `type` is only used when creating and deleting rules and is also immutable.
`type` is the grouping to which the rule applies, it is not returned by `automember_find`
which causes issues when we `compare_args_ipa`. If we dont ignore it will always try modify the automember rule
even if nothing has to be done.